src/action.ts

```diff
@@ -8,37 +8,44 @@
 
 'use strict';
 
-const utils = require('./utils');
+import { Device } from './device';
+import { timestamp } from './utils';
+
+export interface ActionDescription {
+  id: string
+  name: string
+  input: any,
+  status?: string
+  timeRequested?: string
+  timeCompleted?: string
+}
 
 /**
  * An Action represents an individual action on a device.
  */
-class Action {
-  constructor(id, device, name, input) {
-    /**
-     * Initialize the object.
-     *
-     * @param {String} id ID of this action
-     * @param {Object} device Device this action belongs to
-     * @param {String} name Name of the action
-     * @param {Object} input Any action inputs
-     */
-    this.id = id;
-    this.device = device;
-    this.name = name;
-    this.input = input;
-    this.status = 'created';
-    this.timeRequested = utils.timestamp();
-    this.timeCompleted = null;
+export class Action {
+  private status = 'created';
+  private timeRequested = timestamp();
+  private timeCompleted?: string;
+
+  /**
+  * Initialize the object.
+  *
+  * @param {String} id ID of this action
+  * @param {Object} device Device this action belongs to
+  * @param {String} name Name of the action
+  * @param {Object} input Any action inputs
+  */
+  constructor(private id: string, public device: Device, private name: string, private input: any) {
```

Comment:
```suggestion
  constructor(private id: string, public device: Device, private name: string, private input: object) {
```
...basing on JSDoc.

Reply:
`any` is actually right, the doc is wrong.

Reply:
I think are other more places like this.
I'll try to replace most of the `any`s.

Reply:
I'll try to replace it.
But the problem with `object` is you can't access any fields.

Reply:
The problem with object is you can't access any fields.
This might be a bit annoying for the users of the binding.